instrumentation/hibernate/hibernate-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/SessionMethodUtils.java

```diff
@@ -43,7 +43,7 @@
       return null; // No state found. We aren't in a Session.
     }
 
-    int depth = CallDepthThreadLocalMap.incrementCallDepth(SessionMethodUtils.class);
+    int depth = CallDepth.forClass(SessionMethodUtils.class).getAndIncrement();
```

Comment:
Should we have more succinct API? Like `CallDepth.increment(Class)`?

Reply:
Hmm, `CallDepth.increment(Class)` would return an int, correct?
I think that if we use the pattern that Trask posted in this PR there'll be no need for static `increment`/`decrement` methods - we will always have to obtain a `CallDepth` instance exactly once and use it both before and after the original method code. (I'll implement that pattern everywhere if this PR gets merged)

Reply:
Ok, with that pattern the shorter API indeed does not make sense.